crates/polars-python/src/lazyframe/visitor/nodes.rs

```diff
@@ -641,12 +641,59 @@ pub(crate) fn into_py(py: Python<'_>, plan: &IR) -> PyResult<PyObject> {
             contexts: contexts.iter().map(|n| n.0).collect(),
         }
         .into_py_any(py),
+        // Reference get_sink on how to deconstruct the payload
         IR::Sink {
             input: _,
             payload: _,
-        } => Err(PyNotImplementedError::new_err(
-            "Not expecting to see a Sink node",
-        )),
+        } => Sink {
+            input: input.0,
+            payload: match payload {
+                #[allow(unused_variables)]
+                SinkTypeIR::File(FileSinkType {
+                    path,
+                    file_type,
+                    sink_options: _,
+                    cloud_options,
+                }) => {
+                    let path = path.as_ref().as_path();
+                    match &file_type {
+                        #[cfg(feature = "parquet")]
+                        FileType::Parquet(options) => Box::new(ParquetSink::new(
+                            path,
+                            *options,
+                            input_schema.as_ref(),
+                            cloud_options.as_ref(),
+                        )?)
+                            as Box<dyn SinkTrait>,
+                        #[cfg(feature = "ipc")]
+                        FileType::Ipc(options) => Box::new(IpcSink::new(
+                            path,
+                            *options,
+                            input_schema.as_ref(),
+                            cloud_options.as_ref(),
+                        )?) as Box<dyn SinkTrait>,
+                        #[cfg(feature = "csv")]
+                        FileType::Csv(options) => Box::new(CsvSink::new(
+                            path,
+                            options.clone(),
+                            input_schema.as_ref(),
+                            cloud_options.as_ref(),
+                        )?) as Box<dyn SinkTrait>,
+                        #[cfg(feature = "json")]
+                        FileType::Json(options) => Box::new(JsonSink::new(
+                            path,
+                            *options,
+                            input_schema.as_ref(),
+                            cloud_options.as_ref(),
+                        )?)
+                            as Box<dyn SinkTrait>,
+                        #[allow(unreachable_patterns)]
+                        _ => unreachable!(),
```

Comment:
Perhaps we can just do the following to get all the relevant info, serialised as json (which means that if bits change we don't have to make too many matching updates here):

```
        IR::Sink { input, payload } => Sink {
            input: input.0,
            payload: serde_json::to_string(payload)
                .map_err(|err| PyValueError::new_err(format!("{err:?}")))?,
        }
        .into_py_any(py),
```
WDYT?

Reply:
Ah thanks! This works well - I see that is done for the IO readers as well.